services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.service.ts

```diff
@@ -1,124 +1,427 @@
+      programId,
+      programFspConfigurationName,
+    );
+    const existingProperty =
+      await this.getProgramFspConfigurationPropertyOrThrow(
+        config.id,
+        propertyName,
+      );
+
+    existingProperty.value =
+      ProgramFinancialServiceProviderConfigurationMapper.mapPropertyDtoValueToEntityValue(
+        property.value,
+        existingProperty.name,
+      );
+
+    const savedProperty =
+      await this.programFspConfigurationPropertyRepository.save(
+        existingProperty,
+      );
+
+    return ProgramFinancialServiceProviderConfigurationMapper.mapPropertyEntityToDto(
+      savedProperty,
+    );
+  }
+
+  public async deleteProperty({
+    programId,
+    programFspConfigurationName,
+    propertyName,
+  }: {
+    programId: number;
+    programFspConfigurationName: string;
+    propertyName: FinancialServiceProviderConfigurationProperties;
+  }): Promise<void> {
+    const config = await this.getProgramFspConfigurationOrThrow(
+      programId,
+      programFspConfigurationName,
+    );
+    const existingProperty =
+      await this.getProgramFspConfigurationPropertyOrThrow(
+        config.id,
+        propertyName,
+      );
+    await this.programFspConfigurationPropertyRepository.delete({
+      id: existingProperty.id,
+    });
+  }
+
+  private async createProperties(
+    programFspConfigurationId: number,
+    inputProperties: CreateProgramFinancialServiceProviderConfigurationPropertyDto[],
+  ): Promise<ProgramFinancialServiceProviderConfigurationPropertyEntity[]> {
+    const propertiesToSave =
+      ProgramFinancialServiceProviderConfigurationMapper.mapPropertyDtosToEntities(
+        inputProperties,
+        programFspConfigurationId,
+      );
+    return this.programFspConfigurationPropertyRepository.save(
+      propertiesToSave,
+    );
+  }
+
+  private async overwriteProperties(
+    programFspConfigurationId: number,
+    properties: CreateProgramFinancialServiceProviderConfigurationPropertyDto[],
+  ): Promise<ProgramFinancialServiceProviderConfigurationPropertyEntity[]> {
+    // delete all properties
+    await this.programFspConfigurationPropertyRepository.delete({
+      programFinancialServiceProviderConfigurationId: Equal(
+        programFspConfigurationId,
+      ),
+    });
+    // create new properties
+    return await this.createProperties(programFspConfigurationId, properties);
+  }
+
+  private validateFspExists(fspName: string): void {
+    const fsp = FINANCIAL_SERVICE_PROVIDERS.find((fsp) => fsp.name === fspName);
+    if (!fsp) {
+      throw new HttpException(
+        `No fsp found with name ${fspName}`,
+        HttpStatus.NOT_FOUND,
+      );
+    }
+  }
+
+  private validateLabelHasEnglishTranslation(label: any): void {
+    if (!label.en) {
+      throw new HttpException(
+        `Label must have an English translation`,
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+  }
+
+  private async getProgramFspConfigurationOrThrow(
```

Comment:
Move to custom repo.

Reply:
I am not sure if I like that. Same reason as described here: https://github.com/global-121/121-platform/pull/5857#discussion_r1846451539